rust/src/core/polars/mod.rs

```diff
@@ -248,4 +249,42 @@ impl DPNamespace {
         };
         apply_anonymous_function(vec![self.0], args)
     }
+
+    /// Report the argmax or argmin after adding Gumbel noise.
+    ///
+    /// The scale calibrates the level of entropy when selecting an index.
+    ///
+    /// # Arguments
+    /// * `optimize` - Distinguish between argmax and argmin.
+    /// * `scale` - Noise scale parameter for the Gumbel distribution.
+    pub(crate) fn rnm_gumbel(self, optimize: Optimize, scale: Option<f64>) -> Expr {
```

Comment:
surprised scale is an option?

Reply:
It's used to scale the magnitude of the gumbel noise!